models/DataObject/ClassDefinition/Data/Objectbricks.php

```diff
@@ -903,10 +903,15 @@ public function classSaved($class, $params = [])
 
                     foreach ($fieldDefinition as $fd) {
                         //TODO Pimcore 11 remove method_exists call
-                        if (method_exists($fd, 'classSaved')) {
+                        if ($fd instanceof ClassSavedInterface || method_exists($fd, 'classSaved')) {
+                            if (!$fd instanceof ClassSavedInterface) {
+                                trigger_deprecation('pimcore/pimcore', '10.6',
+                                    sprintf('Usage of method_exists is deprecated since version 10.6 and will be removed in Pimcore 11.' .
+                                    'Implement the %s interface instead.', ClassSavedInterface::class));
+                            }
                             // defer creation
                             if (!$fd instanceof DataContainerAwareInterface) {
```

Comment:
We should add this check before `$fd instanceof ClassSavedInterface || method_exists($fd, 'classSaved')` like in the other classes. So we get no deprecation warnings for `DataContainerAwareInterface` classes without `ClassSavedInterface`

Reply:
hope this is what you meant https://github.com/pimcore/pimcore/pull/13791/commits/3ff72c476dfd034e4e0af952fbdd5e73e9dac692

Reply:
Yes 👍🏼